blog/views.py

```diff
@@ -0,0 +1,57 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import generic
+
+from blog.forms import CommentaryForm
+from blog.models import Post, Commentary, User
+
+
+class PostListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.all()
+    template_name = "blog/index.html"
+    paginate_by = 5
+
+
+def post_detail_view(request, pk: int):
```

Comment:
Why don’t you use class based views that inherits from `generic.DetailView`

Reply:
with use function based it's easier to implement a field for comments